### PR TITLE
Cleanup site.js

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

* Updates core-js from v2 to v3
* Moves some functions to the `dom` module (Thus removing some globals)
* Refactors `getParameterByName` to use URLSearchParams from list.

**Issues**

### CODE REVIEW

File: src/scripts/dom.js [hunk 1]
```diff
@@ -262,6 +262,50 @@
...
+     * Gets the value of a parameter in the URL.
+     * @param {*} name Name of the parameter to get.
+     * @param {*} url URL used to get the parameter.
+     */
+    export function getParameterByName(name, url = window.location.href) {
+        const urlObject = new URL(url);
+        const params = new URLSearchParams(urlObject.search);
+
+        return params.get(name);
+    }
```
Comment (MrTimscampi) [inline src/scripts/dom.js:275]: This in particular really needs to be checked. The original is in site.js.

File: src/controllers/dashboard/users/userparentalcontrol.js [hunk 2]
```diff
@@ -261,7 +261,7 @@ define(['jQuery', 'datetime', 'loading', 'libraryMenu', 'globalize', 'listViewSt
     }).on('pageshow', '#userParentalControlPage', function () {
         var page = this;
         loading.show();
-        var userId = getParameterByName('userId');
+        var userId = dom.etParameterByName('userId');
```
Comment (KristupasSavickas) [inline src/controllers/dashboard/users/userparentalcontrol.js:264]:
```suggestion
        var userId = dom.getParameterByName('userId');
```

File: src/scripts/dom.js (hunk 1 above)
Comment (dmitrylyzo) [reply src/scripts/dom.js:275]:
```
url: http://.../jellyfin/web/index.html#!/movies.html?topParentId=b853ad1780b36e82af77f6d2445b7c22
urlObject.search: ''
hash: '#!/movies.html?topParentId=b853ad1780b36e82af77f6d2445b7c22'
```
`params.get('topParentId'): null`

`new URLSearchParams('topParentId=b853ad1780b36e82af77f6d2445b7c22').get('topParentId'): 'b853ad1780b36e82af77f6d2445b7c22'`

Maybe we should just cut of `*?`? `url.replace(/.*\?/, '')`

**Comment by MrTimscampi** [comment]
Closing as this needs to be remade from scratch, given the amount of changes.

**Pull request closed without merging**